File: examples/Phonebook/cpp-server/Phonebook/Major0.cpp

////////////////////////////////////////////////////////
// Types
////////////////////////////////////////////////////////

// Struct: Address
struct Address;

// Struct: Person
struct Person;

// Struct: LookupPerson
struct LookupPerson;

// Struct: LookupPersonByName
struct LookupPersonByName;

// Struct: InsertPerson
struct InsertPerson;

// Struct: Address
struct Address {
   Street street;
   City city;
   Zipcode zipcode;
   State state;
};

// Struct: Person
struct Person {
   Name name;
   Phone home_number;
   Phone cell_number;
   std::optional<Address> address;
   std::vector<PersonId> friends;
};

// Struct: LookupPerson
struct LookupPerson {
   PersonId id;
};

// Struct: LookupPersonByName
struct LookupPersonByName {
   Name name;
};

// Struct: InsertPerson
struct InsertPerson {
   Person person;
};

File: examples/Phonebook/swift-server/Phonebook/Major0.swift

////////////////////////////////////////////////////////
// Types
////////////////////////////////////////////////////////


// Wrap: PersonId
typealias PersonId = String

// Wrap: Name
typealias Name = String

// Wrap: Phone
typealias Phone = String

// Wrap: Street
typealias Street = String

// Wrap: City
typealias City = String

// Wrap: Zipcode
typealias Zipcode = String

// Struct: Address
struct Address {
    let street: Street
    let city: City
    let zipcode: Zipcode
    let state: State
}

// Struct: Person
struct Person {
    let name: Name
    let homeNumber: Phone
    let cellNumber: Phone
    let address: Address?
    let friends: [PersonId]
}

// Struct: LookupPerson
struct LookupPerson {
    let id: PersonId
}

// Struct: LookupPersonByName
struct LookupPersonByName {
    let name: Name
}

// Struct: InsertPerson
struct InsertPerson {
    let person: Person
}

// Enumeration: State
enum State {
    case cA
    case nY
    case tX
    case other(State_other)
}

// Enumeration: State, Tag: other
struct State_other {
    let name: String
}
